### PR TITLE
[js/webgpu] Provide a vectorized algorithm for GroupedConv

### PR DESCRIPTION
**Pull request by qjia7**
### Description
This PR provides a vectorized algorithm for NHWC GroupedConv to improve performance.

The aggregate time of GroupedConv in mobilenetv2-12 becomes ~1ms from ~4ms on Intel Alder Lake machine. About 20% improvement for the whole model.

### CODE REVIEW

**Comment by qjia7** [comment]
@fs-eire @guschmue @satyajandhyala Please take a look, thanks.

**Review by guschmue — DISMISSED** [review]

File: js/web/lib/wasm/jsep/webgpu/ops/conv.ts [hunk 1]
```diff
@@ -136,12 +136,32 @@ const conv2d = (context: ComputeContext, inputs: readonly TensorView[], attribut
   // check attributes
 
   // const hasPreluActivationWeights = false; /* TODO: add support for prelu activation weights */
+  const isChannelsLast = attributes.format === 'NHWC';
   if (attributes.group !== 1) {
-    context.compute(createGroupedConvProgramInfo(inputs, adjustedAttributes));
+    if (isChannelsLast && inputs[1].dims[0] === attributes.group && inputs[1].dims[1] === 1 &&
```
Comment (satyajandhyala) [inline js/web/lib/wasm/jsep/webgpu/ops/conv.ts:141]: Do we need to check unit strides here?

Comment (qjia7) [reply js/web/lib/wasm/jsep/webgpu/ops/conv.ts:141]: `createGroupedConvVectorizeProgramInfo` supports any strides so it won't have error.

Comment (satyajandhyala) [reply js/web/lib/wasm/jsep/webgpu/ops/conv.ts:141]: Thanks for confirming that.

**Review by satyajandhyala — APPROVED** [review]

**Comment by gyagp** [comment]
@fs-eire @guschmue @satyajandhyala any more comments on this?

**Review by fs-eire — APPROVED** [review]

**Comment by gyagp** [comment]
@guschmue @fs-eire Do you think these unsuccessful checks relevant?

**Comment by fs-eire** [comment]
> @guschmue @fs-eire Do you think these unsuccessful checks relevant?

It looks like the unittest are failing:
> [webgpu]Conv - conv - vectorize group - B
> [webgpu]Conv - conv - vectorize group - D

**Comment by fs-eire** [comment]
I am running the unit tests on my local machine to check if CI reports false error.

**Comment by fs-eire** [comment]
> I am running the unit tests on my local machine to check if CI reports false error.

Passed on my local as well. It should be a false error. Let me merge it and watch the main branch.

**Comment by qjia7** [comment]
> > @guschmue @fs-eire Do you think these unsuccessful checks relevant?
> 
> It looks like the unittest are failing:
> 
> > [webgpu]Conv - conv - vectorize group - B
> > [webgpu]Conv - conv - vectorize group - D

I locally tried NV and Intel machines. All cases pass on both of them on latest main. I can't reproduce those failures. Any idea for this?

**Pull request merged**